app/controllers/users_controller.rb

```diff
@@ -1,5 +1,6 @@
 class UsersController < ApplicationController
   before_action :set_user, only: %i[ show liked feed followers following discover ]
+  before_action {authorize @user}
```

Comment:
nit pick: To future proof your code/control for edge cases, consider also authorizing both a user instance OR the user class